web/controllers/rules/pr/no_issue_reference.ex

```diff
@@ -0,0 +1,110 @@
+defmodule Dwylbot.Rules.PR.NoIssueReference do
+  @moduledoc """
+  reassign user if no issue has been referenced
+  """
+  alias Dwylbot.Rules.Helpers
+  @github_api Application.get_env(:dwylbot, :github_api)
+  @rule_name "pr_no_issue_reference"
+
+  def apply?(payload) do
+    payload["action"] in ~w(review_requested assigned)
+  end
+
+  def check(payload, _get_data?, token) do
+
+    authorIsAssigned = payload["pull_request"]["assignees"]
+      |> get_assignees_login
+      |> Enum.member?(payload["pull_request"]["base"]["user"]["login"])
+
+
+    if (!pr_contains_issue_reference?(payload["pull_request"]) && !authorIsAssigned) do
+      %{
+        error_type: @rule_name,
+        actions: [
+          %{
+            remove_assignees: get_assignees_login(payload["pull_request"]["assignees"]),
+            url: "#{payload["pull_request"]["issue_url"]}/assignees"
+          },
+          %{
+            add_labels: ["in-progress"],
+            url: "#{payload["pull_request"]["issue_url"]}/labels"
+          },
+          %{
+            remove_label: "awaiting-review",
+            url: "#{payload["pull_request"]["issue_url"]}/labels"
+          },
+          %{
+            comment: error_message(payload["sender"]["login"]),
+            url: payload["pull_request"]["comments_url"]
+          },
+          %{
+            add_assignees: [ payload["sender"]["login"] ],
+            url: "#{payload["pull_request"]["issue_url"]}/assignees"
+          }
+        ],
+        wait: Helpers.wait(Mix.env, 30_000, 1000, 1),
+        verify: true
```

Comment:
the ```verify``` value means that the rule will be checked again after 30s (1s on dev and 1ms on test). This give the time to the user to edit the comment of the description to add the issue number. However we don't have the condition in the function to get the data of the PR to check again if he description contains the issue number